Modules/xASL_module_ASL.m

```diff
@@ -401,47 +401,53 @@
 %% ========================================================================================================================
 %% 8    Quantification
 
-%% #1543 Here we call xASL_im_MergePWI4D as first function before any quantification (so the user can move it somewhere else/it can be a separate step in a GUI)
-%% This is not part of the quantification, just as HadamardDecoding is also not part of the quantification, it is a resorting/reconcatenation of image & parameter matrices
-
+%% % Use either original or motion estimated ASL4D
+% #1543 Make sure that despiked is used instead of normal ASL in case it exists - to be solved here rather than inside Quantify
+% Use despiked ASL only if spikes were detected and new file has been created
+% Otherwise, despiked_raw_asl = same as original file
+% if ~xASL_exist(x.P.Path_despiked_ASL4D,'file')
+%    x.P.Path_despiked_ASL4D = x.P.Path_ASL4D;
+% end
 
 % Quantification is performed here according to ASL consensus paper (Alsop, MRM 2016)
 % Including PVC
 iState = 8;
 if ~x.mutex.HasState(StateName{iState}) && x.mutex.HasState(StateName{iState-4})
 
+	%% Run merging if needed as first function before any quantification, as quantification itself should disregard the data source
+	if x.modules.asl.bMergingSessions == 1
+		[path_PWI4D, path_PWI4D_Pop] = xASL_im_MergePWI4D(x);
```

Comment:
changing x.P.Path_PWI4D is a bad idea as this is a standardized parameter. We can't redirect it.

Reply:
We do that on several instances, eg with despiked_ASL4D. I understand it's less appropriate, but we need a parameter to propagate to do the QC.

Reply:
With despiked_ASL4D, we create PWI4D based on despikedASL4D instead of based on ASL4D.
So I'm fine creating a new parameter called Path_PWI4D_REALLY_USED_WHATEVER. But not fine changing an automatically generated path parameter that should always point to PWI4D.nii.
so this path_PWI4D stays, but can be saved under a special name into x-struct.

Reply:
Yes, I agree with your solution. We should in fact do this with despikedASL4D as well, instead of changing the fixed x.P.Path_...

We could call it x.P.Path_PWI4D_used
(and similarly x.P.Path_ASL4D_used; but let's keep despikedASL4D for now perhaps;)

alternative = x.P.Path_PWI4D_custom

and we know that x.P.*_used can be overwritten, whereas x.P.* always stays default. Good idea?

Reply:
So right now we have `path_PWI4D`, shall I change that to `path_PWI4D_used`? I'm fine with this.

Just not sure if it's worth it to save it to x.P as we anyway pass this as a function argument to wrp_Quantify and we have to pass it as we sometimes use wrp_Quantify in native and sometimes in standard space. So I'm just not sure if saving to x.P would bring any benefit.

Reply:
Yes let's do path_PWI4D_used, just to clarify that this has a different meaning. I thought about x.P. because we will need this in multiple functions, not only quantification but also visualization, QC, etc.

Reply:
I checked all occurences of x.P.Path_PWI4D and x.P.Pop_Path_PWI4D. And the only relevant place using PWI4D that could be altered here is xASL_wrp_VisualQC_ASL.

So I save stuff to x.P.Path_PWI4D_used and then get it from there for the QC. All other occurences are handled by passing the correct path as function parameter.

Done in 78ed26a4

Reply:
OK great; just note that there may be extra occurrences coming as I will implement more QC steps using this PWI4D (probably in xASL_wrp_VisualQC_ASL). So I'm fine for now.

Please change

if x.modules.asl.bMergingSessions == 1

TO

if x.modules.asl.bMergingSessions

And I don't get why we should have both

path_PWI4D
path_PWI4D_Pop

and 

x.P.Path_PWI4D_used & x.P.Pop_Path_PWI4D_used (note that we need both in visual QC! even it is not there yet)

So I would just use 

x.P.Path_PWI4D_used & x.P.Pop_Path_PWI4D_used

here as well. Adding a description that we also use this in xASL_wrp_Quantify and xASL_wrp_VisualQC.

The less variables the better.

Reply:
1. "if x.modules.asl.bMergingSessions" agree, but that has already been fixed :)
2. "path_PWI4D" - agree, but you looked at a "intermediate" commit. We already use only x.P.Path_PWI4D_used in xASL_module_ASL.
3. Yes - of course, more occurrences are coming - that's fine.
4. Agree - added a comment that the "used" variable is important for xASL_wrp_VisualQC.

4 done in 696ad616